### PR TITLE
feat: WT-2054 Add Loading, Success and Failure screens for transaction

### PR DESCRIPTION
**Pull request by sharonsheah**
# Summary

https://immutable.atlassian.net/browse/WT-2054

### CODE REVIEW

File: packages/internal/bridge/bridge-sample-app/package.json [hunk 1]
```diff
@@ -27,8 +27,8 @@
     "lint": "eslint ./src --ext .ts --max-warnings=0",
     "mapToken": "node --loader ts-node/esm ./src/mapToken.ts",
     "pending": "node --loader ts-node/esm ./src/pending.ts",
-    "status": "node --loader ts-node/esm ./src/status.ts",
     "setFlowRate": "node --loader ts-node/esm ./src/setFlowRate.ts",
+    "status": "node --loader ts-node/esm ./src/status.ts",
```
Comment (imx-mikhala) [inline packages/internal/bridge/bridge-sample-app/package.json:31]: just need to revert this change

File: packages/checkout/widgets-lib/src/context/view-context/BridgeViewContextTypes.ts [hunk 2]
```diff
@@ -59,3 +66,20 @@ interface BridgeClaimWithdrawal extends ViewType {
   type: BridgeWidgetViews.CLAIM_WITHDRAWAL,
   transaction: Transaction
 }
+
+interface BridgeClaimWithdrawalInProgress extends ViewType {
+  type: BridgeWidgetViews.CLAIM_WITHDRAWAL_IN_PROGRESS,
+  data: {
+    transactionResponse: TransactionResponse;
+  }
```
Comment (imx-mikhala) [inline packages/checkout/widgets-lib/src/context/view-context/BridgeViewContextTypes.ts:74]: We should put `transactionResponse: TransactionResponse;` at the root level of this object rather than inside `data`. Data is an any type so we lose intellisense. 

eg:
```
interface BridgeClaimWithdrawalInProgress extends ViewType {
  type: BridgeWidgetViews.CLAIM_WITHDRAWAL_IN_PROGRESS,
  transactionResponse: TransactionResponse;
```

In retrospect I dont think this `data` field on the `ViewType` interface was ever necessary - I know we have it in other places like the `SwapInProgressView` but think this is a mistake

File: packages/checkout/sdk/src/widgets/definitions/events/bridge.ts [hunk 3]
```diff
@@ -28,3 +30,24 @@ export type BridgeFailed = {
...
+  transactionHash: string;
+};
+
+/**
+ * Represents a failed bridge claim withdrawal.
+ * @property {string} reason
+ * @property {number} timestamp
+ */
+export type BridgeClaimWithdrawalFailed = {
+  /** The reason for the failed transaction. */
```
Comment (ZacharyCouchman) [inline packages/checkout/sdk/src/widgets/definitions/events/bridge.ts:49]: Are we able to add the transactionHash to the failure payload as well?

File: packages/checkout/widgets-lib/src/widgets/bridge/BridgeWidget.tsx [hunk 4]
```diff
@@ -215,7 +224,7 @@ export function BridgeWidget({
             />
           )}
           {viewState.view.type === BridgeWidgetViews.TRANSACTIONS && (
-            <Transactions />
+            <Transactions onBackButtonClick={goBackToWalletNetworkSelector} />
```
Comment (ZacharyCouchman) [inline packages/checkout/widgets-lib/src/widgets/bridge/BridgeWidget.tsx:227]: 👏

File: packages/checkout/sdk/src/widgets/definitions/events/bridge.ts [hunk 5]
```diff
@@ -28,3 +30,27 @@ export type BridgeFailed = {
...
+};
+
+/**
+ * Represents a failed bridge claim withdrawal.
+ * @property {string} transactionHash
+ * @property {string} reason
+ * @property {number} timestamp
+ */
+export type BridgeClaimWithdrawalFailed = {
+  /** The hash of the successful transaction. */
```
Comment (imx-mikhala) [inline packages/checkout/sdk/src/widgets/definitions/events/bridge.ts:50]: Need to update text: The hash of the `failed` transaction

**Review by imx-mikhala — DISMISSED** [review]

File: packages/checkout/widgets-lib/src/context/view-context/BridgeViewContextTypes.ts [hunk 6]
```diff
@@ -59,3 +66,18 @@ interface BridgeClaimWithdrawal extends ViewType {
...
+  type: BridgeWidgetViews.CLAIM_WITHDRAWAL_IN_PROGRESS,
+  transactionResponse: TransactionResponse;
+}
+
+export interface BridgeClaimWithdrawalSuccess extends ViewType {
+  type: BridgeWidgetViews.CLAIM_WITHDRAWAL_SUCCESS,
+  transactionHash: string;
+}
+
+interface BridgeClaimWithdrawalFailure extends ViewType {
```
Comment (ZacharyCouchman) [inline packages/checkout/widgets-lib/src/context/view-context/BridgeViewContextTypes.ts:80]: are you passing the transactionHash through if there is a failure? To be added to the event?

File: packages/checkout/widgets-lib/src/components/Transactions/ClaimWithdrawalInProgress.tsx [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+          });
+          return;
+        }
+
+        viewDispatch({
+          payload: {
+            type: ViewActions.UPDATE_VIEW,
+            view: {
+              type: BridgeWidgetViews.CLAIM_WITHDRAWAL_FAILURE,
+              reason: 'Transaction failed',
```
Comment (ZacharyCouchman) [inline packages/checkout/widgets-lib/src/components/Transactions/ClaimWithdrawalInProgress.tsx:51]: I think you'll need to pass the transactionHash in here so that it can be included in the failed event

**Review by ZacharyCouchman — DISMISSED** [review]

**Review by imx-mikhala — DISMISSED** [review]

**Review by ZacharyCouchman — APPROVED** [review]

**Review by imx-mikhala — APPROVED** [review]
